File: 240513/인접하지 않은 2개의 숫자/two-non-adjacent-numbers.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for(int i=0;i<n;i++) {
            arr[i] = sc.nextInt();
        }
        int max = 0;
        for(int i=0;i<n-2;i++) {
            for(int j=i+2;j<n;j++) {
                if(max < arr[i] + arr[j])
                    max = arr[i] + arr[j];
            }
        }
        
        System.out.println(max);
    }
}

File: 240505/인접하지 않은 2개의 숫자/two-non-adjacent-numbers.kt
import java.util.Scanner
import kotlin.math.max

fun main() {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()
    val numbers = IntArray(n) { sc.nextInt() }
    var result = 0

    for (i in 0 until n) {
        for (j in i + 2 until n) {
            result = max(result, numbers[i] + numbers[j])
        }
    }

    print(result)
}